docs/dev/Lua API.rst

```diff
@@ -6623,12 +6623,20 @@ tiletypes
   ``df.tiletype_material``, etc. Returns whether the conversion succeeded.
 
 * ``bool tiletypes_setTile(pos, tiletype_options)`` where
-  the ``tiletype_options`` parameter takes in a table, with any fields
-  matching the available tiletypes options:
-  ``shape``, ``material``, ``special``, ``variant``, ``dig``,
-  ``hidden``, ``light``, ``subterranean``, ``skyview``, ``aquifer``,
-  ``stone_material``, ``vein_type``.
-  Returns whether the conversion succeeded.
+  the ``tiletype_options`` parameter takes in a table, with any fields matching
+  the available tiletypes options. Returns whether the conversion succeeded.
```

Comment:
I'd also mention that an unset field will result in the property not being changed from the existing tile.